### PR TITLE
fix(security): check MIMEType of images and archives of images

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

Fix an issue were it was possible to upload images or archive of images with invalid(s) MIME. 

**Fixes** # MON-5920

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/options/media/images/DB-Func.php [hunk 1]
```diff
@@ -457,3 +457,151 @@ function getListDirectory($directoryNameFilter = null)
...
+ * @return boolean
+ */
+function isCorrectMIMEType($file)
+{
+    $mimeType = mime_content_type($file['tmp_name']);
+
+    if (!preg_match('/(^image\/)|(^application(\/zip)|(\/x-gzip)$)/', $mimeType)) {
+        return false;
+    } else {
+        $dir = sys_get_temp_dir() . '/pendingMedia';
```
Comment (kduret) [inline www/include/options/media/images/DB-Func.php:475]:
```suggestion
        $dir = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'pendingMedia';
```

File: www/include/options/media/images/DB-Func.php [hunk 2]
```diff
@@ -457,3 +457,151 @@ function getListDirectory($directoryNameFilter = null)
...
+    $files = array_diff(scandir($directory), ['..', '.']);
+    $filesInfo = [];
+    foreach ($files as $file) {
+        $filesInfo[] = [
+            'filename' => [
+                'name' => $file,
+                'tmp_name' => $file,
+                'size' => filesize($directory . DIRECTORY_SEPARATOR . $file)
+            ]
+        ];
```
Comment (kduret) [inline www/include/options/media/images/DB-Func.php:604]:
```suggestion
        $filesInfo[] = array(
            'filename' => array(
                'name' => $file,
                'tmp_name' => $file,
                'size' => filesize($directory . DIRECTORY_SEPARATOR . $file)
            )
        );
```

File: www/include/options/media/images/DB-Func.php [hunk 3]
```diff
@@ -457,3 +457,151 @@ function getListDirectory($directoryNameFilter = null)
...
+ * Format all the pending images as an array usable by CentreonImageManager
+ *
+ * @param string $tempDirectory
+ * @return array
+ */
+function getFilesFromTempDirectory($tempDirectory)
+{
+    $directory = sys_get_temp_dir() . DIRECTORY_SEPARATOR . $tempDirectory;
+    $files = array_diff(scandir($directory), ['..', '.']);
+    $filesInfo = [];
```
Comment (kduret) [inline www/include/options/media/images/DB-Func.php:596]:
```suggestion
    $filesInfo = array();
```

File: www/include/options/media/images/DB-Func.php [hunk 4]
```diff
@@ -457,3 +457,151 @@ function getListDirectory($directoryNameFilter = null)
...
+/**
+ * Format all the pending images as an array usable by CentreonImageManager
+ *
+ * @param string $tempDirectory
+ * @return array
+ */
+function getFilesFromTempDirectory($tempDirectory)
+{
+    $directory = sys_get_temp_dir() . DIRECTORY_SEPARATOR . $tempDirectory;
+    $files = array_diff(scandir($directory), ['..', '.']);
```
Comment (kduret) [inline www/include/options/media/images/DB-Func.php:595]:
```suggestion
    $files = array_diff(scandir($directory), array('..', '.'));
```

File: www/include/options/media/images/DB-Func.php [hunk 5]
```diff
@@ -457,3 +457,151 @@ function getListDirectory($directoryNameFilter = null)
...
+
+    if (isset($zip)) {
+        if ($zip->open($filename) === true && $zip->extractTo($dir) === true) {
+            $files = array_diff(scandir($dir), ['..', '.']);
+        } else {
+            return false;
+        }
+    } elseif (isset($tar)) {
+        if ($tar->extractTo($dir) === true) {
+            $files = array_diff(scandir($dir), ['..', '.']);
```
Comment (kduret) [inline www/include/options/media/images/DB-Func.php:572]:
```suggestion
            $files = array_diff(scandir($dir), array('..', '.'));
```

File: www/include/options/media/images/DB-Func.php [hunk 6]
```diff
@@ -457,3 +457,151 @@ function getListDirectory($directoryNameFilter = null)
...
+    /**
+     * Remove Pending images directory to avoid images duplication problems.
+     */
+    if (file_exists($dir)) {
+        removeRecursiveTempDirectory($dir);
+    }
+
+    if (isset($zip)) {
+        if ($zip->open($filename) === true && $zip->extractTo($dir) === true) {
+            $files = array_diff(scandir($dir), ['..', '.']);
```
Comment (kduret) [inline www/include/options/media/images/DB-Func.php:566]:
```suggestion
            $files = array_diff(scandir($dir), array('..', '.'));
```

**Review by kduret — CHANGES_REQUESTED** [review]
needs compatibility with php 5.3 (centos 6)

**Review by kduret — APPROVED** [review]

File: lang/fr/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -1520,6 +1520,12 @@ msgstr "Supprimer"
 msgid "Cancel"
 msgstr "Annuler"
 
+msgid "Invalid Image Format."
+msgstr "Format d'image invalide."
+
+msgid "Images already uploaded."
+msgstr "Images déjà uploadés."
```
Comment (sc979) [inline lang/fr/LC_MESSAGES/messages.po:1527]: uploadé -> téléversée

File: www/include/options/media/images/DB-Func.php [hunk 8]
```diff
@@ -457,3 +457,151 @@ function getListDirectory($directoryNameFilter = null)
...
+        $filesInfo[] = array(
+            'filename' => array(
+                'name' => $file,
+                'tmp_name' => $file,
+                'size' => filesize($directory . DIRECTORY_SEPARATOR . $file)
+            )
+        );
+    }
+    return $filesInfo;
+}
```
Comment (sc979) [inline www/include/options/media/images/DB-Func.php:607]:
```suggestion
}

```

**Review by sc979 — APPROVED** [review]

File: lang/fr/LC_MESSAGES/messages.po (hunk 7 above)
Comment (kduret) [reply lang/fr/LC_MESSAGES/messages.po]: I'm not a big fan of `téléversée`
Target people are technical, and know better `uploaded` than `téléversées`, even if there are french
What do you think @cgagnaire & @lpinsivy ?

Comment (cgagnaire) [reply lang/fr/LC_MESSAGES/messages.po]: We could also use `chargée` ?

**Review by kduret — APPROVED** [review]

**Pull request merged**